src/coverlet.collector/DataCollector/CoverletCoverageDataCollector.cs

```diff
@@ -0,0 +1,179 @@
+        /// Initializes data collector
+        /// </summary>
+        /// <param name="configurationElement">Configuration element</param>
+        /// <param name="events">Events to register on</param>
+        /// <param name="dataSink">Data sink to send attachments to test platform</param>
+        /// <param name="logger">Test platform logger</param>
+        /// <param name="environmentContext">Environment context</param>
+        public override void Initialize(
+            XmlElement configurationElement,
+            DataCollectionEvents events,
+            DataCollectionSink dataSink,
+            DataCollectionLogger logger,
+            DataCollectionEnvironmentContext environmentContext)
+        {
+            if (this.eqtTrace.IsInfoEnabled)
+            {
+                this.eqtTrace.Info("Initializing {0} with configuration: '{1}'", CoverletConstants.DataCollectorName, configurationElement?.OuterXml);
+            }
+
+            // Store input variables
+            this.events = events;
+            this.configurationElement = configurationElement;
+            this.dataSink = dataSink;
+            this.dataCollectionContext = environmentContext.SessionDataCollectionContext;
+            this.logger = new TestPlatformLogger(logger, this.dataCollectionContext);
+
+            // Register events
+            this.events.SessionStart += this.OnSessionStart;
+            this.events.SessionEnd += this.OnSessionEnd;
+        }
+
+        /// <summary>
+        /// Disposes the data collector
+        /// </summary>
+        /// <param name="disposing">Disposing flag</param>
+        protected override void Dispose(bool disposing)
+        {
+            this.eqtTrace.Verbose("{0}: Disposing", CoverletConstants.DataCollectorName);
+
+            // Unregister events
+            if (this.events != null)
+            {
+                this.events.SessionStart -= this.OnSessionStart;
+                this.events.SessionEnd -= this.OnSessionEnd;
+            }
+
+            // Remove vars
+            this.events = null;
+            this.dataSink = null;
+            this.coverageManager = null;
+
+            base.Dispose(disposing);
+        }
+
+        /// <summary>
+        /// SessionStart event handler
+        /// </summary>
+        /// <param name="sender">Sender</param>
+        /// <param name="sessionStartEventArgs">Event args</param>
+        private void OnSessionStart(object sender, SessionStartEventArgs sessionStartEventArgs)
+        {
+            this.eqtTrace.Verbose("{0}: SessionStart received", CoverletConstants.DataCollectorName);
+
+            try
+            {
+                // Get coverlet settings
+                IEnumerable<string> testModules = this.GetTestModules(sessionStartEventArgs);
+                var coverletSettingsParser = new CoverletSettingsParser(this.eqtTrace);
+                var coverletSettings = coverletSettingsParser.Parse(this.configurationElement, testModules);
```

Comment:
Please avoid `var` when type is not `clear` here and everywhere
```
var coverletSettingsParser = new CoverletSettingsParser(this.eqtTrace); <-- ok type is clear
var coverletSettings = coverletSettingsParser.Parse(this.configurationElement, testModules); <-- type is no clear
CoverletSettings coverletSettings = coverletSettingsParser.Parse(this.configurationElement, testModules); <-- type is clear
```

Reply:
Avoid `var` when type is `clear`, right?

Reply:
Not sure to follow...but my english is bad :smile: so I try to explain...to me is easy understand code flow when I recognize types, in case of `var type = method.Get()` is not clear the type of `var type` so in this cases I prefer `string type = method.Get()` by contrast `var type = new MyType()` is clear enough.
My sense of "clear" is a synonym of "follow code without go to definition with VS to understand what is the used type"

Reply:
OMG it's time to some holiday...you're right 😅 it's ok never mind.

Reply:
Interesting I missed that, maybe because I'm used to apply that rule too strictly.

Reply:
Haha. I think I've got your point. 😄 
When you know the type, don't use `var`. I'll make sure of that. :P

Reply:
Done